test/python/utils.py

```diff
@@ -20,11 +28,271 @@ def tarantool_version():
+    'app/roles/custom.lua',
+    'test',
+    'test/helper',
+    'test/integration',
+    'test/unit',
+    'tmp',
+    'tmp/.keep',
+    project_name + '-scm-1.rockspec',
+    'tarantool',
+    'tarantoolctl',
+    'VERSION',
+    'ignored',  # special folder for test work cartridge ignore
+    'ignored/asterisk'
+])
+
+original_rocks_content = set([
+    '.rocks',
+    '.rocks/share/tarantool/rocks/manifest',
+    '.rocks/share/tarantool/rocks/' + project_name,
+    '.rocks/share/tarantool/rocks/' + project_name + '/scm-1',
+    '.rocks/share/tarantool/rocks/' + project_name + '/scm-1/rock_manifest',
+    '.rocks/share/tarantool/rocks/' + project_name + '/scm-1/' + project_name + '-scm-1.rockspec',
+
+    '.rocks/bin/luatest',
+    '.rocks/share/tarantool/checks.lua',
+    '.rocks/share/tarantool/luarocks/test/luatest.lua',
+    '.rocks/share/tarantool/luatest',
+    '.rocks/share/tarantool/rocks/checks',
+    '.rocks/share/tarantool/rocks/luatest',
+])
+
+
+def assert_dir_contents(files_list, skip_tarantool_binaries=False):
+    without_rocks = {x for x in files_list if not x.startswith('.rocks')}
+
+    file_tree = original_file_tree
+    if skip_tarantool_binaries or not tarantool_enterprise_is_used():
+        file_tree = {x for x in file_tree if x not in ['tarantool', 'tarantoolctl']}
+
+    assert file_tree == without_rocks
+    assert all(x in files_list for x in original_rocks_content)
+
+
+def assert_filemode(filepath, filemode):
+    filepath = os.path.join('/', filepath)
+
+    if filepath == os.path.join('/usr/share/tarantool/', project_name, 'VERSION'):
+        assert filemode & 0o777 == 0o644
+    elif filepath.startswith('/etc/systemd/system/'):
+        assert filemode & 0o777 == 0o644
+    elif filepath.startswith('/usr/lib/tmpfiles.d/'):
+        assert filemode & 0o777 == 0o644
+    elif filepath.startswith('/usr/share/tarantool/'):
+        # a+r for files, a+rx for directories
+        required_bits = 0o555 if os.path.isdir(filepath) else 0o444
+        assert filemode & required_bits == required_bits
+
+
+def assert_filemodes(basedir):
+    known_dirs = set([
+        'etc', 'etc/systemd', 'etc/systemd/system',
+        'usr', 'usr/share', 'usr/share/tarantool',
+        'usr/lib', 'usr/lib/tmpfiles.d'
+    ])
+    filenames = recursive_listdir(basedir) - known_dirs
+
+    for filename in filenames:
+        # we don't check fileowner here because it's set in postinst script
+
+        # check filemode
+        if filename.startswith(os.path.join('usr/share/tarantool/', project_name, '.rocks')):
+            continue
+
+        # get filestat
+        file_stat = os.stat(os.path.join(basedir, filename))
+        filemode = file_stat.st_mode
+        assert_filemode(filename, filemode)
+
+
+def validate_version_file(distribution_dir):
+    original_keys = [
+        'TARANTOOL',
+        project_name,
+        # default app dependencies
+        'luatest',
+        'cartridge',
+        # known cartridge dependencies
```

Comment:
`ddl` for new versions...

I'm not sure that we should hardcode it in test - it could be different for different versions

Reply:
I was thinking about it and tried to list application directory and then filter out some files, but it turned out to be not so easy. I decided that this change needs separated PR in the future.

Now, the `cartridge` version is hardcoded in application template, and you should fix this files list after changing it.

Reply:
I suggest you to file an issue to fix it. If cartridge version is hardcoded - ok. For now it's ok

Reply:
https://github.com/tarantool/cartridge-cli/issues/96